antarest/storage/business/watcher.py

```diff
@@ -1,11 +1,68 @@
+import logging
+import threading
+from pathlib import Path
+from time import time, sleep
+from typing import List
+from filelock import FileLock  # type: ignore
+
 from antarest.common.config import Config
+from antarest.login.model import Group
+from antarest.storage.model import StudyFolder
 from antarest.storage.service import StorageService
 
 
+logger = logging.getLogger(__name__)
+
+
 class Watcher:
+    LOCK = Path("watcher")
+    DELAY = 1
+
     def __init__(self, config: Config, service: StorageService):
         self.service = service
         self.config = config
 
-    def init(self) -> None:
-        return None
+        self.thread = (
+            threading.Thread(target=self._loop)
+            if Watcher._get_lock()
+            else None
+        )
+
+    def start(self) -> None:
+        if self.thread:
+            self.thread.start()
+
+    @staticmethod
+    def _get_lock() -> bool:
+        with FileLock(f"{Watcher.LOCK}.lock"):
+            start = (
+                int(f"0{Watcher.LOCK.read_text()}")
+                if Watcher.LOCK.exists()
+                else 0
+            )
+            now = int(time())
+            if now - start > Watcher.DELAY:
+                Watcher.LOCK.write_text(str(now))
+                logger.info("Watcher get lock")
+                return True
+            else:
```

Comment:
i don't understand. if the second thread enter this lock within the watcher delay (1 second is large), it guess it will also start ?

Reply:
It starts only if date stored in file, is older than DELAY

Reply:
oh yes, my bad